agogosml_cli/cli/templates/{{cookiecutter.PROJECT_NAME_SLUG}}/agogosml-ci-pipeline.yml

```diff
@@ -0,0 +1,126 @@
+resources:
+- repo: self
+
+variables:
+  container_registry: '{{cookiecutter.AZURE_CONTAINER_REGISTRY}}/'
+
+trigger:
+  branches:
+    include:
+    - master
+  paths:
+    include:
+    - '**/agogosml'
+    - '**/input_reader'
+    - '**/output_writer'
+
+jobs:
+- job: Phase_1
+  displayName: Agogosml-Build-CI
+  condition: succeeded()
+  pool:
+    vmImage: 'Ubuntu 16.04'
+
+  steps:
+  - task: Docker@1
+    displayName: 'Build an image'
+    inputs:
+      azureSubscriptionEndpoint: '{{cookiecutter.SUBSCRIPTION_ID}}'
+      azureContainerRegistry: {{cookiecutter.AZURE_CONTAINER_REGISTRY}}
+      dockerFile: '**/agogosml/Dockerfile.agogosml'
+      imageName: 'agogosml:$(Build.BuildId)'
+
+  - bash: |  
+       if [ $(Build.SourceBranchName) = "master" ]; then
+         echo "Since this is a merge build, we are creating a new 'latest' tag"
+         docker tag $(container_registry)agogosml:$(Build.BuildId) $(container_registry)agogosml:latest
+         echo "Done"
+       else
+         echo "Not a merge build, hence skipping 'latest' tag"
+       fi
+
+    displayName: 'Tag latest tag'
+
```

Comment:
any reason we don't rely on the in-box Docker task to "tag" --- a push to a registry, since the checksums would all be the same, results in just a 'taggnig' of the same image layer.

```
  - task: Docker@1
    displayName: 'Push the latest image'
    inputs:
      azureSubscriptionEndpoint: 'myContainerRegistryRGServiceConnection'
      azureContainerRegistry: 'myContainerRegistry.azurecr.io'
      command: 'push'
      imageName: "$(Build.Repository.Name):latest"
```

Reply:
@eladiw can you please chime in here? I don't remember why we had to tag separately.